### PR TITLE
Filter "guests only" pages to restore backwards compatibility

### PR DESCRIPTION
**Pull request by leofeyer**
Another follow-up PR to #3143.

Since we are no longer using the `find*WithoutGuests()` methods, we have to filter the "guests only" pages in the menu modules. This also allows us to trigger a deprecation warning, which is desirable.

### CODE REVIEW

File: core-bundle/src/Resources/contao/modules/Module.php [hunk 1]
```diff
@@ -317,6 +317,13 @@ protected function renderNavigation($pid, $level=1, $host=null, $language=null)
 				trigger_deprecation('contao/core-bundle', '4.12', 'Using a tabindex value greater than 0 has been deprecated and will no longer work in Contao 5.0.');
 			}
 
+			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
+			if ($objSubpage->guests && !$objSubpage->protected && !$user)
```
Comment (ausi) [inline core-bundle/src/Resources/contao/modules/Module.php]:
```suggestion
			if ($objSubpage->guests && !$objSubpage->protected && $user)
```

File: core-bundle/src/Resources/contao/modules/ModuleBooknav.php [hunk 2]
```diff
@@ -205,6 +205,13 @@ protected function getBookPages($intParentId, $groups, $time)
 		{
 			$objPage->loadDetails();
 
+			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
+			if ($objPage->guests && !$objPage->protected && !$user)
```
Comment (ausi) [inline core-bundle/src/Resources/contao/modules/ModuleBooknav.php]:
```suggestion
			if ($objPage->guests && !$objPage->protected && $user)
```

File: core-bundle/src/Resources/contao/modules/ModuleCustomnav.php [hunk 3]
```diff
@@ -95,6 +95,13 @@ protected function compile()
 		{
 			$objModel->loadDetails();
 
+			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
+			if ($objModel->guests && !$objModel->protected && !$user)
```
Comment (ausi) [inline core-bundle/src/Resources/contao/modules/ModuleCustomnav.php]:
```suggestion
			if ($objModel->guests && !$objModel->protected && $user)
```

File: core-bundle/src/Resources/contao/modules/ModuleQuicklink.php [hunk 4]
```diff
@@ -94,6 +94,13 @@ protected function compile()
 		{
 			$objSubpage->loadDetails();
 
+			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
+			if ($objSubpage->guests && !$objSubpage->protected && !$user)
```
Comment (ausi) [inline core-bundle/src/Resources/contao/modules/ModuleQuicklink.php]:
```suggestion
			if ($objSubpage->guests && !$objSubpage->protected && $user)
```

File: core-bundle/src/Resources/contao/modules/ModuleQuicknav.php [hunk 5]
```diff
@@ -132,6 +132,13 @@ protected function getQuicknavPages($pid, $level=1, $host=null)
 				$objSubpage->domain = $host;
 			}
 
+			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
+			if ($objSubpage->guests && !$objSubpage->protected && !$user)
```
Comment (ausi) [inline core-bundle/src/Resources/contao/modules/ModuleQuicknav.php]:
```suggestion
			if ($objSubpage->guests && !$objSubpage->protected && $user)
```

File: core-bundle/src/Resources/contao/modules/Module.php [hunk 6]
```diff
@@ -313,6 +313,13 @@ protected function renderNavigation($pid, $level=1, $host=null, $language=null)
 				trigger_deprecation('contao/core-bundle', '4.12', 'Using a tabindex value greater than 0 has been deprecated and will no longer work in Contao 5.0.');
 			}
 
+			// Hide the page if it is not protected and only visible to guests (backwards compatibility)
+			if ($objSubpage->guests && !$objSubpage->protected && $user)
```
Comment (aschempp) [inline core-bundle/src/Resources/contao/modules/Module.php]: there's no `$user`?

**Review by aschempp — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/modules/Module.php (hunk 6 above)
Comment (ausi) [reply core-bundle/src/Resources/contao/modules/Module.php]: Line 281 and 285?

Comment (aschempp) [reply core-bundle/src/Resources/contao/modules/Module.php]: not sure what you're looking at, but this has been removed by my frontend group voter and probably a rebase

Bildschirmfoto 2021-07-05 um 08 32 32

Comment (ausi) [reply core-bundle/src/Resources/contao/modules/Module.php]: You are right, I looked at the file before the rebase/merge https://github.com/contao/contao/blob/79a8dcff1c6805b15ff3820cf4f6c6c854a83ddd/core-bundle/src/Resources/contao/modules/Module.php#L281-L285

So this needs to be fixed indeed.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/modules/Module.php]: Fixed in d3d00c9e709d7de8084d293f16f2ca25f247e1fd. Is it correct this way or should I have checked `ROLE_MEMBER` instead?

Comment (aschempp) [reply core-bundle/src/Resources/contao/modules/Module.php:318]: never check the user, always ask a voter (= check `ROLE_MEMBER`)

**Review by aschempp — CHANGES_REQUESTED** [review]
never check the user, always ask a voter (= check `ROLE_MEMBER`)

**Comment by leofeyer** [comment]
Fixed in d38daaa5098054bdd8a5953db45ccab35a1db2d0.

**Review by ausi — APPROVED** [review]

**Pull request merged**